doc/auto_techsupport_and_coredump_mgmt.md

```diff
@@ -68,14 +76,59 @@ The naming format and compression is governed by the script `/usr/local/bin/core
 
 Where `<comm>` value in the command name associated with a process. comm value of a running process can be read from `/proc/[pid]/comm` file
 
-## 4. Schema Additions
+## 4. Memory usage based techsupport invocation
+
+If the following condition resolves to true:
+```
+(mem_usage > mem_usage_threshold || ${container}_mem_usage > ${container}_mem_usage_threshold) || mem_free <= mem_free_threshold
+```
+
+where ```mem_usage```  is total system memory used (MemAvailable from /proc/meminfo),
+```mem_usage_threshold```  configured threshold, (100 - available_mem_threshold),
+
+```${container}_mem_usage```  used memory by $container ("docker stats --no-stream --format {{.MemUsage}}" $container),
+
+```${container}_mem_usage_threshold```  configured memory threshold for $container, (100 - ${container}available_mem_threshold), 
+
+```mem_free``` is the total minus mem usage, ```mem_free_threshold``` - mem free threshold.
+
+the SONiC techsupport is automatically generated.
+
+```mem_free_threshold``` is there to invoke dump when there is quite small amount of memory left that is needed to successfully execute "show techsupport". This is going to be 200 Mb by default, as at least 80-90 Mb takes "show techsupport" execution.
+
+The check will be implemented as a script that is ran by monit periodically:
+
+```
+check program mem_checker with path "/usr/bin/mem_threshold_check"
+    if status != 0 for 10 times within 20 cycles then exec /usr/local/bin/mem_threshold_check_handler"
+```
+
+The action is going to be ran only once the mem_check script detects memory usage above threshold.

```

Comment:
>once [](http://example.com/codeflow?start=35&length=4)

Do you mean once after each bootup? Considering some fluctuations and finally a memory exhausting, you will keep only the first fluctuation "show techsupport" file?

Reply:
I mean that the techsupport will run once if the script returns non zero status code for 10 times within 20 cycles. If memory usage is constantly too high for e.g 100 cycles, only 1 dump will be taken. If lets say the memory usage fluctuates, in example, 10 cycles too high in a row and then goes down for 10 cycles and then again up - there is a protection against too frequent auto tech dumps already present in the auto tech infrastructure.